### PR TITLE
operator: Build and publish images for docker hub

### PR DESCRIPTION
**Pull request by shwetaap**
Signed-off-by: Shweta Padubidri <spadubid@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -550,6 +550,14 @@ endif
 	$(call push,loki-build-image,latest)
 
 
+# loki-operator
+loki-operator-image:
+	$(SUDO) docker build -t $(IMAGE_PREFIX)/loki-operator:$(IMAGE_TAG) -f operator/Dockerfile .
+loki-operator-image-cross:
+	$(SUDO) $(BUILD_OCI) -t $(IMAGE_PREFIX)/loki-operator:$(IMAGE_TAG) -f operator/Dockerfile.cross .
+loki-operator-push: loki-operator-image-cross
+        $(SUDO) $(PUSH_OCI) $(IMAGE_PREFIX)/loki-operator:$(IMAGE_TAG)
```
Comment (chaudum) [inline Makefile:559]: This indentation looks weird ;-)

File: .drone/drone.jsonnet [hunk 2]
```diff
@@ -282,6 +294,27 @@ local lambda_promtail(tags='') = pipeline('lambda-promtail') {
...
+      settings+: {
+        dry_run: true,
+      },
+    },
+  ] + [
+    // publish for tag or main
+    docker_operator(arch, 'loki-operator') {
+      depends_on: ['image-tag'],
+      when: condition('include').tagMain,
+      settings+: {},
```
Comment (chaudum) [inline .drone/drone.jsonnet:312]: Could be omitted.

File: operator/Dockerfile.cross [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+RUN go mod download
+
+# Copy the go source
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
+COPY internal/ internal/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -a -o manager main.go
```
Comment (chaudum) [inline operator/Dockerfile.cross:24]: I don't know too much about cross platform builds, but as I understood, `GOARCH` and `GOARM` environment variables must be provided.
At least the values for them are copied into the Dockerfile in line 8.

```suggestion
RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on GOARCH=$(cat /goarch) GOARM=$(cat /goarm) go build -a -o manager main.go
```

File: Makefile (hunk 1 above)
Comment (shwetaap) [reply Makefile]: Done

**Review by chaudum — APPROVED** [review]

**Pull request merged**